### ISSUE TITLE
A11y: Focus state and label for hamburger menu

### ISSUE THREAD

**Issue by robdodson** [pbakaus/goweb#4]
Consider replacing the `.nav-icon` div with a button to get better keyboard support and automatic focusability. It should have a keyboard handler as well as an aria-label that says something like `Open Menu`.

Also, you'll want to prevent focus from entering the drawer menu when it's closed. The easiest way to do this is to set it to `visibility: hidden` or `display: none`. Similarly when it's opened you'll want to move focus into it, and trap focus so it can't escape the menu until it's closed. Ideally you also shouldn't be able to escape it using a screen reader but in practice that's a lot harder than just trapping focus (it requires setting aria-hidden on everything that isn't the drawer menu).

Menu should also support closing when the user presses `Esc`.
